### ISSUE TITLE
Translations for the release 3.12 wanted 

### ISSUE THREAD

**Issue by mgrojo** [sqlitebrowser/sqlitebrowser#2124]
We are in the process of releasing our next stable version (3.12.0) of 'DB Browser for SQLite' with many new features and corrections.

Once again we kindly ask our users to help us in making DB4S better by translating it to their languages or updating the translation of the last release (when available).

The *.ts files are already updated on the master branch, and we are not going to modify the strings unless needed to include some correction.

You can read [brief indications here](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Translations) about how to do it. I'll be happy to assist you in the process, in case you had any doubt.

I'll ping here all the translators who have helped us in the past, so they get a notification. Please, report back if you want to help again. When there are more than one volunteers, they can share out the work and/or review each other translations. Happy translation!

lang | complete | translators
-- | -- | --
ar_SA | <li>[x] Yes</li> | @safaalfulaij
cs | <li>[ ] </li> | @MrJiggers 
de | <li>[x] Yes</li> | @FriedrichFroebel
en_GB | <li>[ ] </li> | @glawrence 
es_ES | <li>[x] Yes</li> | @lulol, @mgrojo
fr | <li>[x] Yes</li> | @mvt91
it | <li>[x] Yes</li> | @Feyn-Man, @GortiZ6 
ja (new) | <li>[x] Yes</li> | @hrwatahiki 
ko_KR | <li>[ ] </li> | @progh2, @ChangJoo-Park 
pl | <li>[x] Yes</li> | @bartlomiej-przymus, @wojnilowicz 
pt_BR | <li>[x] Yes</li> | @bernardosulzbach
ru | <li>[ ] </li> | @schdub, @zvova7890, @VanDerSam 
tr | <li>[x] Yes</li> | @firateski 
uk_UA | <li>[ ] </li> | @keedhost 
zh | <li>[ ] Needs review</li> | @bssthu, @mercury233 
zh_TW | <li>[ ] </li> | @PeterDaveHello 

Source string changes after translation started:

* 1d1afaad218dfea9d9cdcf83821b3c5b33963ebb
* https://github.com/sqlitebrowser/sqlitebrowser/commit/fe88463cbad5476ca6555759eb70dee430d89f63#diff-1c0b64226a129d35f02dc0049484cd6b
* https://github.com/sqlitebrowser/sqlitebrowser/commit/498efd7a5c0ceb43ebcf1233bee4609a201f0e40#diff-8b5b29ce11f2efcd1c93eb6cdf67ce89
* 39047109a65e325c1a2135a85a372efaa315316d
* https://github.com/sqlitebrowser/sqlitebrowser/commit/498efd7a5c0ceb43ebcf1233bee4609a201f0e40#diff-8b5b29ce11f2efcd1c93eb6cdf67ce89
* https://github.com/sqlitebrowser/sqlitebrowser/commit/29db270e809980d3a36ee6c30fd30bd12a5a5cd7

When we're near the final release build, we should check this list of "source string changes ..." in case some languages need those done.

**Comment by FriedrichFroebel** [comment]
I will have a look at updating the German translation in the next days.

**Comment by bernardosulzbach** [comment]
I will make some time for this in the next 24 hours.

**Comment by GortiZ6** [comment]
I'll try to complete it at the beginning of next week.

Il Ven 14 Feb 2020, 21:00 Bernardo Sulzbach <notifications@github.com> ha
scritto:

**Comment by hrwatahiki** [comment]
Thanks for accepting my pull request.
I will update the Japanese file to 3.12 by 2020-02-18 (UTC).

**Comment by mvt91** [comment]
Hello,

I'll try to do it this week-end. I've to reinstall the software to do it.
Best regards
Michel

----- Mail d'origine -----
De: Manuel <notifications@github.com>
À: sqlitebrowser/sqlitebrowser <sqlitebrowser@noreply.github.com>
Cc: mvt91 <mvt91@free.fr>, Mention <mention@noreply.github.com>
Envoyé: Fri, 14 Feb 2020 20:28:45 +0100 (CET)
Objet: [sqlitebrowser/sqlitebrowser] Translations for the release 3.12 wanted  (#2124)

We are in the process of releasing our next stable version (3.12.0) of 'DB Browser for SQLite' with many new features and corrections.

Once again we kindly ask our users to help us in making DB4S better by translating it to their languages or updating the translation of the last release (when available).

The *.ts files are already updated on the master branch, and we are not going to modify the strings unless needed to include some correction.

You can read [brief indications here](https://github.com/sqlitebrowser/sqlitebrowser/issues/182#issuecomment-70413095) about how to do it. I'll be happy to assist you in the process, in case you had any doubt.

I'll ping here all the translators who have helped us in the past, so they get a notification. Please, report back if you want to help again. When there are more than one volunteers, they can share out the work and/or review each other translations. Happy translation!

lang | translators
-- | --
ar_SA |  @safaalfulaij
cs |  @MrJiggers 
de |  @FriedrichFroebel
en_GB |  @glawrence 
es_ES |  @lulol, @mgrojo
fr |  @mvt91
it | @Feyn-Man, @GortiZ6 
ko_KR |  @progh2, @ChangJoo-Park 
pl |  @bartlomiej-przymus, @wojnilowicz 
pt_BR |  @bernardosulzbach
ru |  @schdub, @zvova7890, @VanDerSam 
tr |  @firateski 
uk_UA |  @keedhost 
zh |  @bssthu, @mercury233 
zh_TW |  @PeterDaveHello 

-- 
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub:
https://github.com/sqlitebrowser/sqlitebrowser/issues/2124

**Comment by firateski** [comment]
I'm planning to complete Turkish translations in the next 3-4 days. Do you have any tool suggestions to easily manage translations?

**Comment by bernardosulzbach** [comment]
> I'm planning to complete Turkish translations in the next 3-4 days. Do you have any tool suggestions to easily manage translations?

Besides Qt Linguist? If that's the case, then no. I think most people translating are just using Linguist.

**Comment by mvt91** [comment]
Hi Manuel,

I just finish the job :)
Going to bed right now :)

Best regards

Michel

----- Mail d'origine -----
De: Manuel <notifications@github.com>
À: sqlitebrowser/sqlitebrowser <sqlitebrowser@noreply.github.com>
Cc: mvt91 <mvt91@free.fr>, Mention <mention@noreply.github.com>
Envoyé: Fri, 14 Feb 2020 20:28:45 +0100 (CET)
Objet: [sqlitebrowser/sqlitebrowser] Translations for the release 3.12 wanted  (#2124)

We are in the process of releasing our next stable version (3.12.0) of 'DB Browser for SQLite' with many new features and corrections.

Once again we kindly ask our users to help us in making DB4S better by translating it to their languages or updating the translation of the last release (when available).

The *.ts files are already updated on the master branch, and we are not going to modify the strings unless needed to include some correction.

You can read [brief indications here](https://github.com/sqlitebrowser/sqlitebrowser/issues/182#issuecomment-70413095) about how to do it. I'll be happy to assist you in the process, in case you had any doubt.

I'll ping here all the translators who have helped us in the past, so they get a notification. Please, report back if you want to help again. When there are more than one volunteers, they can share out the work and/or review each other translations. Happy translation!

lang | translators
-- | --
ar_SA |  @safaalfulaij
cs |  @MrJiggers 
de |  @FriedrichFroebel
en_GB |  @glawrence 
es_ES |  @lulol, @mgrojo
fr |  @mvt91
it | @Feyn-Man, @GortiZ6 
ko_KR |  @progh2, @ChangJoo-Park 
pl |  @bartlomiej-przymus, @wojnilowicz 
pt_BR |  @bernardosulzbach
ru |  @schdub, @zvova7890, @VanDerSam 
tr |  @firateski 
uk_UA |  @keedhost 
zh |  @bssthu, @mercury233 
zh_TW |  @PeterDaveHello 

-- 
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub:
https://github.com/sqlitebrowser/sqlitebrowser/issues/2124

**Comment by mvt91** [comment]
Here is the file... I don't know where to push it :(

Please remove the .zip extension
[sqlb_fr.ts.zip](https://github.com/sqlitebrowser/sqlitebrowser/files/4207542/sqlb_fr.ts.zip)

**Comment by justinclift** [comment]
Thanks @mvt91, just added that in daedace373dfe1dcd268fe6c16dacf4b0c76d399. :smile:

As a thought, there were two spelling mistakes in our source (english) sentences, fixed a few hours ago.

* https://github.com/sqlitebrowser/sqlitebrowser/commit/1d1afaad218dfea9d9cdcf83821b3c5b33963ebb
* https://github.com/sqlitebrowser/sqlitebrowser/commit/fe88463cbad5476ca6555759eb70dee430d89f63#diff-1c0b64226a129d35f02dc0049484cd6b

Would you be ok to grab the [.ts file](https://github.com/sqlitebrowser/sqlitebrowser/blob/master/src/translations/sqlb_fr.ts) again (it has the corrected sentences in it now), and translate those two as well?

**Comment by justinclift** [comment]
@SafaAlfulaij @hrwatahiki Same question as above... there were two spelling mistakes fixed in our source text a few hours ago.  That will mean your translations aren't **quite** 100% any more.

Any interest in fixing those as well? :smile:

**Comment by firateski** [comment]
> > I'm planning to complete Turkish translations in the next 3-4 days. Do you have any tool suggestions to easily manage translations?
> 
> Besides Qt Linguist? If that's the case, then no. I think most people translating are just using Linguist.

I just heard about the Qt Linguist. Qt Linguist looking fine, thank you.

**Comment by mvt91** [comment]
I Justin,

The updates are done.
Where I can left the file ? as early this morning ?)

Thanks

Michel

----- Mail d'origine -----
De: Justin Clift <notifications@github.com>
À: sqlitebrowser/sqlitebrowser <sqlitebrowser@noreply.github.com>
Cc: mvt91 <mvt91@free.fr>, Mention <mention@noreply.github.com>
Envoyé: Sat, 15 Feb 2020 07:17:40 +0100 (CET)
Objet: Re: [sqlitebrowser/sqlitebrowser] Translations for the release 3.12 wanted  (#2124)

Thanks @mvt91, just added that in daedace373dfe1dcd268fe6c16dacf4b0c76d399. :smile:

As a thought, there were two spelling mistakes in our source (english) sentences, fixed a few hours ago.

* https://github.com/sqlitebrowser/sqlitebrowser/commit/1d1afaad218dfea9d9cdcf83821b3c5b33963ebb
* https://github.com/sqlitebrowser/sqlitebrowser/commit/fe88463cbad5476ca6555759eb70dee430d89f63#diff-1c0b64226a129d35f02dc0049484cd6b

Would you be ok to grab the [.ts file](https://github.com/sqlitebrowser/sqlitebrowser/blob/master/src/translations/sqlb_fr.ts) again (it has the corrected sentences in it now), and translate those two as well?

**Comment by mvt91** [comment]
Hi Firat,

I use this :
https://www.softpedia.com/get/Others/Home-Education/Qt-Linguist.shtml

Best regards
----- Mail d'origine -----
De: Firat Eski <notifications@github.com>
À: sqlitebrowser/sqlitebrowser <sqlitebrowser@noreply.github.com>
Cc: mvt91 <mvt91@free.fr>, Mention <mention@noreply.github.com>
Envoyé: Sat, 15 Feb 2020 11:46:04 +0100 (CET)
Objet: Re: [sqlitebrowser/sqlitebrowser] Translations for the release 3.12 wanted  (#2124)

> > I'm planning to complete Turkish translations in the next 3-4 days. Do you have any tool suggestions to easily manage translations?
> 
> Besides Qt Linguist? If that's the case, then no. I think most people translating are just using Linguist.

I just heard about the Qt Linguist. Qt Linguist looking fine, thank you.

**Comment by mvt91** [comment]
Hi Justin (bis),

Here is the file updated, as early this morning, please remove the .zip extension.
Best regards
[sqlb_fr.ts.zip](https://github.com/sqlitebrowser/sqlitebrowser/files/4208418/sqlb_fr.ts.zip)

**Comment by justinclift** [comment]
@mvt91 Thanks, added in d8b25e0b8ec9ec94b87d48f826e130f58debe4b5. :smile:

**Comment by hrwatahiki** [comment]
@justinclift Thank you. Now I've download the new files and I'm updating them.

**Comment by SafaAlfulaij** [comment]
Puzzle strings, and untranslatable string:
https://github.com/sqlitebrowser/sqlitebrowser/blob/master/src/AboutDialog.cpp#L21
Please user proper placeholders for variables.

**Comment by justinclift** [comment]
@hrwatahiki Thanks. :smile:

@SafaAlfulaij That's a bit confusing.  What should that example be instead? :smile:

**Comment by justinclift** [comment]
Oops.  Accidentally closed this. :wink:

**Comment by firateski** [comment]
> @hrwatahiki Thanks. 😄
> 
> @SafaAlfulaij That's a bit confusing. What should that example be instead? 😄

If I understood @SafaAlfulaij correctly, he pointed out that we need parameterized strings in 'AboutDialog.cpp' (which is referenced by Safa), because in some languages ​​we use a different word order than English to make sentences.

For example, in the following code block string concatenation is okay for English but not for Turkish: 

` tr("Built for ") + QSysInfo::buildAbi()...`

English: "Built for __X__"
Turkish: "__X__ için derlendi" <- We need to put X before the predicate for Turkish.

In short, parameterized strings allow us to make a sentence in a valid order.

**P.S:** I'm able to fix this problem by using another form of the sentence like "Derleme Sürümü: X", so it's not a huge problem for Turkish, at least for now.

**Comment by bernardosulzbach** [comment]
> In short, parameterized strings allow us to make a sentence in a valid order.

Some Portuguese translations could be more "natural" if the source strings were more encompassing, too.

**Comment by mgrojo** [comment]
Yes, you're right. We'll try to review that in following changes. I've committed the change of two of the strings in the About dialog. I've left unchanged the more simple ones, which I suppose have been worked around. Is it now in better condition? 498efd7a5c0ceb43ebcf1233bee4609a201f0e40

**Comment by bernardosulzbach** [comment]
Should I update the Portuguese translation now or is it OK as-is for 3.12 @mgrojo?

**Comment by justinclift** [comment]
We seem to be finding strings for improvement on the way through, so I've just added a new "Source string changes after translation started" list here in the issue (first post).

That way we can check for "things that have changed" before we do the release build, so everyone doesn't have to create a new translation every time there's a new string change.  eg Do it in one hit.

Hopefully that's an improvement. :smile:

**Comment by bernardosulzbach** [comment]
I agree, @justinclift. Ping me right before the release then and I will get my translation to 100% again.

**Comment by justinclift** [comment]
@bernardosulzbach Will do. :smile:

**Comment by firateski** [comment]
I updated Turkish translation and created a PR. Please ping me as well, if any sources change from now on. @justinclift

**Comment by justinclift** [comment]
Thanks @firateski, just merged your Turkish translation PR.  And yep, will ping you nearer our release if any sources change. :smile:

Am kind of planning on pinging everyone really, if the sources change. :grin:

**Comment by keedhost** [comment]
Will be done soon

**Comment by SafaAlfulaij** [comment]
When is the release? I'd like to know how much time I have since I'd go over the translation and improve some bits.

And, if you don't like Qt Linguist, you can use Lokalize, it supports ts :)

**Comment by justinclift** [comment]
Lokalize seems to have a bug in it, whereby it changes the (html?) encoding of entities.

It's not a big problem though, as running `lupdate` on the output of Lokalize seems to change them back ok + keep the translations.

**Comment by justinclift** [comment]
> When is the release?

We'll make the first alpha release "soon".  Personally I was kind of thinking "when we have the translations", but it also depends on whether there is outstanding stuff we want in the release.  I'm not 100% sure of that.  @MKleusberg @mgrojo @scottfurry ?

**Comment by justinclift** [comment]
@GortiZ6 Thanks. :smile:

Took a look through the updated .ts file, and it all seems ok.  Structurally at least, as I don't read Italian. :wink:

**Comment by FriedrichFroebel** [comment]
I just pushed the updates for the German translation. There were two possible issues with the language files I saw:

1. In line 1659 of `MainWindow.cpp`, the string

       Select the action to apply to the dropped file(s). <br/>Note: only 'Import' will process more than one file.

    should be translated. Qt Linguist does not allow me to finish the corresponding translations for the singular and plural case, as it is complaining about missing placeholders (`%n/%Ln`) inside my translation. But the placeholder is missing inside the source string already.

2. In line 526 of `TableBrowser.ui`, we have the string

       <html><head/><body><p>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Clicking this button navigates up to the end in the table view above.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</p></body></html>

   In this case, the source string looks somehow strange as the HTML markup includes the same markup again, but this time escaped. All the other HTML-based source strings do not seem to employ this behaviour. At the moment my translation omits the escaped HTML markup.

**Comment by SafaAlfulaij** [comment]
> should be translated. Qt Linguist does not allow me to finish the corresponding translations for the singular and plural case, as it is complaining about missing placeholders (%n/%Ln) inside my translation. But the placeholder is missing inside the source string already.

You can still "finish" it and it will appear as a yellow check mark. If you are certin that this will work (and you've tested it), then no problem.

**Comment by mgrojo** [comment]
There is a problem with all the translations, but it seems a problem in Qt, because the ts files seem ok.

![imagen](https://user-images.githubusercontent.com/13812910/74870906-d4c40680-535a-11ea-89f5-ebf3240baffe.png)

I've seen the same problem in the Spanish, German and Italian translations, so I suppose it happens in everyone. My Qt version is Qt 5.9.5. Does anyone else reproduce it? Any idea about the reason?

**Comment by mgrojo** [comment]
It is probably due to the way in which the header is contructed. I tried to call the translation later, but it still fails to get the translation:

```diff
diff --git a/src/RemoteModel.cpp b/src/RemoteModel.cpp
index a8f5b62a..f50274fd 100644
--- a/src/RemoteModel.cpp
+++ b/src/RemoteModel.cpp
@@ -7,7 +7,7 @@
 using json = nlohmann::json;
 
 // The header list is a list of column titles
-const static std::vector<QString> headerList = {QObject::tr("Name"), QObject::tr("Commit"), QObject::tr("Last modified"), QObject::tr("Size")};
+const static std::vector<const char*> headerList = {"Name", "Commit", "Last modified", "Size"};
 
 RemoteModelItem::RemoteModelItem(RemoteModelItem* parent) :
     m_parent(parent),
@@ -255,7 +255,7 @@ QVariant RemoteModel::headerData(int section, Qt::Orientation orientation, int r
         return QAbstractItemModel::headerData(section, orientation, role);
 
     // Return header string depending on column
-    return headerList.at(static_cast<size_t>(section));
+    return tr(headerList.at(static_cast<size_t>(section)));
 }
 
 int RemoteModel::rowCount(const QModelIndex& parent) const

```

Maybe we need to use a conventional approach?

**Comment by justinclift** [comment]
@MKleusberg ? ^^^

**Comment by MKleusberg** [comment]
It should be working now. I haven't spend too much time investigating this but I would think it's because of the ```static``` keyword which makes this evaluated before the language settings are loaded.

I've also fixed the second point in @FriedrichFroebel's comment above. This was definitely an error and even visible to the user:
![Screenshot_20200220_115258](https://user-images.githubusercontent.com/3153504/74928749-63c43380-53da-11ea-928a-eb0d0485cf67.png)
I'm not sure if the 'up' in the text is really idiomatic either. Also the translations need to be updated after my change.

**Comment by GortiZ6** [comment]
Updated italian on pull request #2141 ;)

**Comment by justinclift** [comment]
Thanks @GortiZ6, just merged it. :smile:

**Comment by mercury233** [comment]
https://github.com/sqlitebrowser/sqlitebrowser/blob/29db270/src/translations/sqlb_zh.ts#L2192
the <p> in the first line is not closed
(although it will be ignored by most html parser)

**Comment by mercury233** [comment]
![image](https://user-images.githubusercontent.com/13391795/75257690-49e28080-5820-11ea-9f58-ae4cc3ea0156.png)
how to translate those buttons?

**Comment by justinclift** [comment]
> how to translate those buttons?

I **think** the standard dialog buttons translations are part of the main Qt project.  So they need to be translated in the Qt library. :frowning:

**Comment by MKleusberg** [comment]
Yes, @justinclift is right. We should ship two translation files for each language. One is named sqlb_zh.qm, the other qt_zh.qm. The second one contains translations for these buttons and is provided with the Qt framework. Since they are obviously in English in your case we definitely need to check if this file is included in our package (@justinclift ?) and if so figure out why it is not loaded.

**Comment by MKleusberg** [comment]
At least in the zip release for Windows I could see the right files. I have just pushed a fix which tries a different file name which might work better :smile:

**Comment by justinclift** [comment]
Awesome.  Hopefully that fixes it. :smile:

**Comment by SafaAlfulaij** [comment]
Well, Arabic translations from Qt library works properly before the commit, and I can see the qt_zh files there (which should work normally, but don't).
Actually I don't see a qtbase_zh file, so the new commit might be even breaking stuff. Explanation:
Basically in Qt5, translation files changed. In 4, qt_xx had everything in it.
In 5, qt_xx changed to reference other files like qtbase, qtmultimedia, qtscript, etc.

Some languages ported the files, and some didn't. But even though, qt_xx should just work fine if it has all translations.
I also checked Portogues (since it also has everything in qt_pt.qml file), and those strings didn't get translated.

I'm not sure where is the problem. I'll check the new nightly build and inform if something broke.

**Comment by MKleusberg** [comment]
Thanks, @SafaAlfulaij! I think you're right. [Here](https://doc.qt.io/qt-5/linguist-programmers.html#deploying-translations)'s the documentation on that. The ```lconvert``` example in the docs might be worth trying too.

**Comment by MKleusberg** [comment]
Looks like the abbreviations of the languages we use doesn't always match the [ones used by Qt](https://doc.qt.io/qt-5/linguist-programmers.html#deploying-translations):

Our language code | Qt's language code
------------ | -------------
ar_SA | ar
cs | cs
de | de
en_GB | 
es_ES | es
fa | fa
fr | fr
it | it
ja | ja
ko_KR | ko
pl | pl
pt_BR | pt
ru | ru
tr | 
uk_UA | uk
zh_TW | zh_TW
zh | zh_CN

So I guess, this is what we should do to fix this issue:
1) Rename ```zh``` to ```zh_CN```.
2) Change our code to look for a Qt translation file which is named like ```qt_x.qm``` when the language code is something like ```x_y```, except for languages like ```zh_y```.

What do you think?

**Comment by scottfurry** [comment]
On 2020-02-26 9:10 a.m., Martin Kleusberg wrote:
>
> Looks like the abbreviations of the languages we use doesn't always 
> match the ones used by Qt 
> <https://doc.qt.io/qt-5/linguist-programmers.html#deploying-translations>:
>
....
>
> So I guess, this is what we should do to fix this issue:
>
>  1. Rename |zh| to |zh_CN|.
>  2. Change our code to look for a Qt translation file which is named
>     like |qt_x.qm| when the language code is something like |x_y|,
>     except for languages like |zh_y|.
>
> What do you think?
>
I wouldn't worry too much about "Qt uses this".
My suggestion is to keep using ISO-639(language code) ISO-3166(country 
code) codes are set as file names.
Qt Linguist should "pick up" filenames/languages "automagically". See 
this nice explainer from Oracle for VirtualBox (which also uses Qt).

**Comment by mgrojo** [comment]
For my system, the current approach is working, even though the system Qt file is named '`/usr/share/qt5/translations/qt_es.qm`, so I guess in Linux it is automatically loaded.

I don't mind if our files are renamed, but given the case for Chinese, can't we just look for `qt_x_y.qm` and when it is not OK, look for `qt_x.qm` for any language?

**Comment by scottfurry** [comment]
On 2020-02-26 12:46 p.m., Scott Furry wrote:
> See this nice explainer from Oracle for VirtualBox (which also uses Qt).
Totally oblivious to the fact I didn't include the link. My bad...

https://www.virtualbox.org/wiki/Translating

FWIW, qgis (also uses Qt) has translation files that follow the format...
qgis_[ISO 639 language code (lower case)].ts

English: qgis_en.ts
German : qgis_de.ts
etc...

One thing I would suggest is to go from least specific to most specific.
For example:

English (least)             _en.ts
English(UK - specific)      _en_uk.ts
English(US - specific)      _en_us.ts
English(Canadian)           _en_ca.ts

French(least)               _fr.ts
French(Canadian - specific) _fr_ca.ts

Ensure, at minimum, there is a "least specific" language file (e.g. 
_en.ts and _fr.ts) and more specific ( e.g. _en_uk.ts _fr_ca.ts ) where 
necessary.
So long as we follow a clear and consistent format, I don't see a problem.

**Comment by mercury233** [comment]
I guess the zh translation of qt must be manually added, see https://github.com/wisaly/qtbase_zh

**Comment by justinclift** [comment]
We should check if the Chinese (zh) translation is working now, after the filename rename.  And probably the existing ones too, just to make sure it didn't break them.

That aside, it looks like some of our translations aren't going to be updated for this release.  Oh well. :face_with_thermometer:

I'm inclined to create a new `v3.12.x` branch from master in the very near future, and then build our initial Alpha release.  Are there substantial pieces of code we're waiting for, or need reviewing first?

**Comment by mercury233** [comment]
The dialog buttons of qt are still not translated in Chinese.

**Comment by justinclift** [comment]
@mercury233 Yeah.  Personally, I'm not sure if this is a general problem with Qt itself not having translations, or if it's something else.

**Comment by mgrojo** [comment]
To all our translators:

We are pleased to announce that we have just published the [release candidate 1](https://sqlitebrowser.org/blog/first-release-candidate-for-3-12-0/) for our 3.12.0 release. Unless some new bug appears, we don't expect to change any translatable string in the [v3.12.x branch](https://github.com/sqlitebrowser/sqlitebrowser/tree/v3.12.x). It's the perfect time to complete the translations and enjoy DB4S 3.12 in your language.

Some numbers to know the effort in advance:

Language | Remaining string
--------------|-----------------------
ar_SA | 6
cs | 563
ru | 271
de | 6
fr | 8
zh | 67
zh_TW | 961
pl | 8
pt_BR | 8
en_GB | 919
es_ES | 6
ko_KR | 439
tr | 7
uk_UA | 638
it |5

**Comment by bernardosulzbach** [comment]
Hi, Manuel. I don't know if this is the first time you posted the "remaining strings count" with the call-to-action or not, but thank you very much. This helps a lot with scheduling. I should open a PR within the next 12 hours.

**Comment by mgrojo** [comment]
> Hi, Manuel. I don't know if this is the first time you posted the "remaining strings count" with the call-to-action or not, but thank you very much. This helps a lot with scheduling. I should open a PR within the next 12 hours.

It's the first time. I'll do it again in the future, if it's helpful.

**Comment by mvt91** [comment]
Hi Manuel,

I missed something... Damned.

I'll get the file and check it in a few days.

Best regards
Michel
----- Mail d'origine -----
De: Manuel <notifications@github.com>
À: sqlitebrowser/sqlitebrowser <sqlitebrowser@noreply.github.com>
Cc: mvt91 <mvt91@free.fr>, Mention <mention@noreply.github.com>
Envoyé: Tue, 09 Jun 2020 20:57:01 +0200 (CEST)
Objet: Re: [sqlitebrowser/sqlitebrowser] Translations for the release 3.12 wanted  (#2124)

To all our translators:

We are pleased to announce that we have just published the [release candidate 1](https://sqlitebrowser.org/blog/first-release-candidate-for-3-12-0/) for our 3.12.0 release. Unless some new bug appears, we don't expect to change any translatable string in the [v3.12.x branch](https://github.com/sqlitebrowser/sqlitebrowser/tree/v3.12.x). It's the perfect time to complete the translations and enjoy DB4S 3.12 in your language.

Some numbers to know the effort in advance:

Language | Remaining string
--------------|-----------------------
ar_SA | 6
cs | 563
ru | 271
de | 6
fr | 8
zh | 67
zh_TW | 961
pl | 8
pt_BR | 8
en_GB | 919
es_ES | 6
ko_KR | 439
tr | 7
uk_UA | 638
it |5

**Comment by SafaAlfulaij** [comment]
Nothing serious for Arabic (all are shortcuts and this empty string):
`        <location filename="../RemotePushDialog.ui" line="49"/>`

**Comment by mercury233** [comment]
From my view, in the `zh` translation, only 62 strings are remained, and all of them are the document of SqlLexer, which I can't translate.
![image](https://user-images.githubusercontent.com/13391795/84212623-edfee780-aaf0-11ea-8b8c-595051bd33e7.png)

but you get 67 strings, where are the rest 5?

**Comment by justinclift** [comment]
@mercury233 For the count of 62 strings, is that in the `master` branch, or the `v3.12.x` branch? :smile:

**Comment by mercury233** [comment]
@justinclift it is the `master` branch. Why does `v3.12.x` is ahead of master? I'll translate the new strings  in `v3.12.x` later.

**Comment by justinclift** [comment]
`v3.12.x` probably isn't ahead of `master`, they just have slightly different strings now. :smile:

As a priority thing, translating `v3.12.x` would be better than `master` if you need to choose just one.  We'll be making the official 3.12.0 release this weekend, from that branch. :smile:

**Comment by mercury233** [comment]
It seems that `v3.12.x` didn't include my last pr (https://github.com/sqlitebrowser/sqlitebrowser/pull/2231), can you please merge it?

**Comment by justinclift** [comment]
Oops.  Ok, just merged it. :smile:

**Comment by mercury233** [comment]
Now the only problem of zh translation is the dialog buttons of qt, as I mentioned in https://github.com/sqlitebrowser/sqlitebrowser/issues/2124#issuecomment-590901102

**Comment by justinclift** [comment]
Yeah.  I don't know if there's any way we can fix that. :frowning:

**Comment by justinclift** [comment]
Actually, maybe we could just add the zh translations to our own copy of the Qt translation files on our build server...

**Comment by mercury233** [comment]
can you try this? https://github.com/wisaly/qtbase_zh

**Comment by justinclift** [comment]
Ok, I've just added the .qm from that repo to the 64-bit Qt translations folder, and just started our nightly build script.

In about 40 minutes, an updated version of our Win64 build should be on the nightlies server.  When that new build is ready, are you able to check if it improves things?

&nbsp; &nbsp; https://nightlies.sqlitebrowser.org/latest/

**Comment by mercury233** [comment]
It don't work, and the string I just commited to `v3.12.x` didn't appear
![image](https://user-images.githubusercontent.com/13391795/84248032-74d8b200-ab3b-11ea-827f-e88c7804dd27.png)

**Comment by justinclift** [comment]
Ahhh.  That build pulled from the `master` branch, it's expected it won't get changes to the `v3.12.x` one.

I was hoping those Qt buttons at the bottom would change though. :frowning:

**Comment by mvt91** [comment]
Hi Justin,

Here is the new french translation file. I hope everything is correct :)

You'll find 2 files : the ts file and ts file zipped.

Best regards

Michel

----- Mail d'origine -----
De: Justin Clift <notifications@github.com>
À: sqlitebrowser/sqlitebrowser <sqlitebrowser@noreply.github.com>
Cc: mvt91 <mvt91@free.fr>, Mention <mention@noreply.github.com>
Envoyé: Wed, 10 Jun 2020 11:33:08 +0200 (CEST)
Objet: Re: [sqlitebrowser/sqlitebrowser] Translations for the release 3.12 wanted  (#2124)

Ahhh.  That build pulled from the `master` branch, it's expected it won't get changes to the `v3.12.x` one.

I was hoping those Qt buttons at the bottom would change though. :frowning:

**Comment by justinclift** [comment]
@mvt91 Sorry, but that didn't work.  GitHub ignores attachments in email reply messages. :frowning:

You'll probably need to use the web interface instead. :smile:

**Comment by justinclift** [comment]
@mvt91 Not sure about that error.  Looking at the .ts file, I **think** the error is mistaken.  It seems to be thinking the source string has placeholders in it - which it doesn't.  As you say, it probably doesn't matter.  Hopefully. :smile:

For the .zip files, I just applied the first one to the `master` branch as that'll improve the translation there.  Then applied the new .zip file to the `v3.12.x` branch.  It all seems to be happy, so Well Done. :smile:

**Comment by mvt91** [comment]
Hi Justin,

I tried to find a solution but it seems (stackoverflow) it's a little bug in the Qt linguist version I used.

For me it's Ok.

Best regards

Michel
----- Mail d'origine -----
De: Justin Clift <notifications@github.com>
À: sqlitebrowser/sqlitebrowser <sqlitebrowser@noreply.github.com>
Cc: mvt91 <mvt91@free.fr>, Mention <mention@noreply.github.com>
Envoyé: Thu, 11 Jun 2020 07:47:53 +0200 (CEST)
Objet: Re: [sqlitebrowser/sqlitebrowser] Translations for the release 3.12 wanted  (#2124)

@mvt91 Not sure about that error.  Looking at the .ts file, I **think** the error is mistaken.  It seems to be thinking the source string has placeholders in it - which it doesn't.  As you say, it probably doesn't matter.  Hopefully. :smile:

For the .zip files, I just applied the first one to the `master` branch as that'll improve the translation there.  Then applied the new .zip file to the `v3.12.x` branch.  It all seems to be happy, so Well Done. :smile:

**Comment by justinclift** [comment]
@mvt91 No worries. :smile:

**Comment by mvt91** [comment]
No No... thanks
----- Mail d'origine -----
De: Justin Clift <notifications@github.com>
À: sqlitebrowser/sqlitebrowser <sqlitebrowser@noreply.github.com>
Cc: mvt91 <mvt91@free.fr>, Mention <mention@noreply.github.com>
Envoyé: Thu, 11 Jun 2020 12:32:20 +0200 (CEST)
Objet: Re: [sqlitebrowser/sqlitebrowser] Translations for the release 3.12 wanted  (#2124)

@mvt91 No worries. :smile:

**Comment by MKleusberg** [comment]
Closing this as version 3.12 is finally released :smile:

**Comment by justinclift** [comment]
Good point. :smile:

Thanks heaps to everyone for the translations.  The approach this time around (initial translations done, then a final "touch up" of the translations for the release candidate) seems to work out well.

We should probably do the same thing again next time. :smile:
